VAMobile/src/screens/HomeScreen/ProfileScreen/ContactInformationScreen/EditPhoneNumberScreen/EditPhoneNumberScreen.tsx

```diff
@@ -90,9 +88,46 @@ const EditPhoneNumberScreen: FC<IEditPhoneNumberScreen> = ({ navigation, route }
 
   const onSave = (): void => {
     const onlyDigitsNum = getNumbersFromString(phoneNumber)
-    const numberId = phoneData && phoneData.id ? phoneData.id : 0
+    const id = phoneData.id || 0
 
-    dispatch(editUsersNumber(phoneType, onlyDigitsNum, extension, numberId, saveSnackbarMessages, ScreenIDTypesConstants.EDIT_PHONE_NUMBER_SCREEN_ID))
+    let phoneDataPayload: PhoneData = {
+      areaCode: onlyDigitsNum.substring(0, 3),
+      countryCode: '1',
+      phoneNumber: onlyDigitsNum.substring(3),
+      phoneType,
+    }
+
+    if (extension) {
+      phoneDataPayload = {
+        ...phoneDataPayload,
+        extension,
+      }
+    }
+
+    const phoneNumberExists = (contactInformation || {})[PhoneTypeToFormattedNumber[phoneType as PhoneType] as keyof UserContactInformation]
+
+    if (phoneNumberExists) {
+      phoneDataPayload = {
+        ...phoneDataPayload,
+        id,
+      }
+
+      const update = () =>
+        updatePhoneNumber(phoneDataPayload, {
+          onSuccess: () => showSnackBar(saveSnackbarMessages.successMsg, dispatch, undefined, true, false, true),
+          onError: () => showSnackBar(saveSnackbarMessages.errorMsg, dispatch, update, false, true, true),
+        })
+
+      update()
```

Comment:
If this is the only place the function is used, probably don't need a function

Reply:
It's also used in the `showSnackBar` function, because it gets used as the retry function on errors. That's the main reason I wrapped that logic in a separate function, to avoid needing to pass in the parameters to the mutation function twice. I have [variations](https://github.com/department-of-veterans-affairs/va-mobile-app/pull/6733/files#diff-4b0dfaf60714e697abefaf28a855abc1523660a9c20139bce60f4d65a3ba2e54R124-R128) of this where I store the mutate options in a separate variable, and just pass that into the functions. Mainly experimenting with different styles to see which one feels the most natural